xarray/tests/test_backends.py

```diff
@@ -1535,6 +1535,77 @@ def test_keep_chunksizes_if_no_original_shape(self) -> None:
                 ds["x"].encoding["chunksizes"], actual["x"].encoding["chunksizes"]
             )
 
+    def test_preferred_chunks_is_present(self) -> None:
+        ds = Dataset({"x": [1, 2, 3]})
+        chunksizes = (2,)
+        ds.variables["x"].encoding = {"chunksizes": chunksizes}
+
+        with self.roundtrip(ds) as actual:
+            assert actual["x"].encoding["preferred_chunks"] == {"x": 2}
+
+    @requires_dask
+    def test_auto_chunking_is_based_on_disk_chunk_sizes(self) -> None:
+        x_size = y_size = 1000
+        y_chunksize = y_size
+        x_chunksize = 10
+
+        with self.create_chunked_file(
+            (1, y_size, x_size), (1, y_chunksize, x_chunksize)
+        ) as tmp_file:
+            with dask.config.set({"array.chunk-size": "100KiB"}):
+                with open_dataset(tmp_file, chunks="auto", engine=self.engine) as ds:
+                    t_chunks, y_chunks, x_chunks = ds["image"].data.chunks
+                    assert all(np.asanyarray(y_chunks) == y_chunksize)
+                    # Check that the chunk size is a multiple of the file chunk size
+                    assert all(np.asanyarray(x_chunks) % x_chunksize == 0)
+
+    @requires_dask
+    def test_base_chunking_uses_disk_chunk_sizes(self) -> None:
+        x_size = y_size = 1000
+        y_chunksize = y_size
+        x_chunksize = 10
+
+        with self.create_chunked_file(
+            (1, y_size, x_size), (1, y_chunksize, x_chunksize)
+        ) as tmp_file:
+            with open_dataset(tmp_file, chunks={}, engine=self.engine) as ds:
+                for chunksizes, expected in zip(
+                    ds["image"].data.chunks, (1, y_chunksize, x_chunksize)
+                ):
+                    assert all(np.asanyarray(chunksizes) == expected)
+
+    @contextlib.contextmanager
+    def create_chunked_file(self, array_shape, chunk_sizes) -> None:
```

Comment:
```suggestion
    def create_chunked_file(self, array_shape: tuple[int, int, int], chunk_sizes: tuple[int, int, int]) -> Generator[str, None, None]:
```
I haven't checked it though, but it should bring you in the right direction.